### PR TITLE
feature/admin-crud-reward

### PR DESCRIPTION
**Pull request by kajawitek**
Hi Nerds!

I created new model: reward and CRUD reward for admin user. New reward specs have been added.

### CODE REVIEW

File: app/controllers/admins/rewards_controller.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    end
+
+    def new
+      @reward = Reward.new
+    end
+
+    def create
+      @reward = Reward.new(reward_params)
+
+      if @reward.title.empty?
```
Comment (oskarlakner) [inline app/controllers/admins/rewards_controller.rb:22]: this is not needed at all and in fact an overkill. Google please for Rails validations. A model should validate the object and if there is a missing title the action should render a view with an instance of the object, then in the view you should render these errors to the user.

File: app/views/admins/rewards/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<p id="notice"><%= notice %></p>
+
+<h1>New Reward</h1>
+
+<%= render 'form', reward: @reward %>
+
+<%= link_to 'Back', admins_rewards_path %>
+
```
Comment (oskarlakner) [inline app/views/admins/rewards/new.html.erb:8]: empty lines - please check your PR carefully before sending it to code review

File: config/routes.rb [hunk 3]
```diff
@@ -14,6 +14,7 @@
     root 'pages#dashboard'
     resources :employees
     resources :company_values
+    resources :rewards
```
Comment (oskarlakner) [inline config/routes.rb:19]: same

File: spec/models/reward_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+RSpec.describe Reward, type: :model do
+  let!(:reward) { create(:reward) }
+
+  it 'is valid with valid attributes' do
+    expect(reward).to be_valid
+  end
+
+  context 'when title is empty' do
+    before do
+      reward.title = nil
```
Comment (oskarlakner) [inline spec/models/reward_spec.rb:12]: check shoulda matchers gem

**Review by oskarlakner — APPROVED** [review]
you can do better :P

File: app/controllers/admins/rewards_controller.rb (hunk 1 above)
Comment (kajawitek) [reply app/controllers/admins/rewards_controller.rb:22]: Thanks! Now I know more about validations and it is and will be very useful knowledge :)

**Comment by kajawitek** [comment]
> you can do better :P

True! I will :)

**Pull request merged**